python/arcticdb/version_store/_store.py

```diff
@@ -2805,10 +2830,20 @@ def defragment_symbol_data(self, symbol: str, segment_size: Optional[int] = None
         in the future. This API will allow overriding the setting as well.
         """
 
-        if self._lib_cfg.lib_desc.version.write_options.bucketize_dynamic:
+        proto_cfg = self._lib_cfg.lib_desc.version.write_options
+        if proto_cfg.bucketize_dynamic:
             raise ArcticDbNotYetImplemented(f"Support for library with 'bucketize_dynamic' ON is not implemented yet")
 
-        result = self.version_store.defragment_symbol_data(symbol, segment_size, prune_previous_versions)
+        # All other methods use prune_previous_version, but also support prune_previous_versions here in case
+        # anyone is relying on it
+        if prune_previous_versions is not None:
+            prune_previous_version = prune_previous_versions
+        else:
+            prune_previous_version = self.resolve_defaults(
```

Comment:
The `existing_value` arg on `resolve_defaults` seems to be designed for this purpose, rather than the `if is not None` logic up in its caller

Reply:
Fair, I'd confused myself over `prune_previous_versions` vs `prune_previous_version`. Used `existing_value` now